packages/components/src/components/linear-progress/linear-progress-container.jsx

```diff
@@ -2,61 +2,46 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { LinearProgress } from './linear-progress.jsx';
 
-class LinearProgressContainer extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            timeout: props.timeout / 1000,
-            total_ticks: Math.round(props.timeout / 1000),
-            current_tick: Math.round(props.timeout / 1000),
-        };
-    }
+const LinearProgressContainer = ({ timeout, action, render, className }) => {
+    const [timeout_state, setTimeout] = React.useState(timeout);
+    const [current_tick, setCurrentTick] = React.useState(Math.round(timeout / 1000));
+    const total_ticks = React.useState(Math.round(timeout / 1000));
 
-    get progress() {
-        return 100 - Math.round((this.state.current_tick / this.state.total_ticks) * 100);
-    }
+    const getProgress = () => 100 - Math.round((current_tick / total_ticks) * 100);
 
-    get remaining() {
-        return this.state.timeout >= 0 ? this.state.timeout : 0;
-    }
+    const getRemaining = () => (timeout >= 0 ? timeout : 0);
 
-    makeProgress = () => {
-        this.setState({
-            current_tick: this.state.current_tick - 1,
-            timeout: this.state.timeout - 1,
-        });
+    const makeProgress = () => {
+        setCurrentTick(current_tick - 1);
+        setTimeout(timeout_state - 1);
     };
 
-    run = () => {
-        this.props.action();
+    const run = () => {
```

Comment:
Might as well remove `run` `func` and use `action()` directly on line L32.

Reply:
Removed as suggested